### ISSUE TITLE
MultiCanvas canvases not filled

### ISSUE THREAD

**Issue by chinshr** [katspaugh/wavesurfer.js#691]
I had a chance to take a look and test the `MultiCanvas` drawer. Great job @chrisparton1991! I really appreciate your effort.

I ran into a problem when peaks were loaded externally. Using `fillParent`. The issue appears when the waveform per canvas is not filled correctly, which appears to be a scaling issue, which I appear to have fixed, here is the code:

```js
    drawLineToContext: function (ctx, index, peaks, absmax, halfH, scale, offsetY) {
        if (!ctx) { return; }

        var length = ~~(peaks.length / this.canvases.length / 2);
        var first = index * length,
            last = first + length + 1;

        scale = 1;
        if (this.params.fillParent && this.width != length) {
            scale = ctx.canvas.width / length;
        }

       ...
```

If time permits, I will work on a pull request.

**Comment by chrisparton1991** [comment]
Hi @chinshr, thanks for the feedback :)

I'm a bit busy at the moment myself. I put your code through the tests I added to my original PR (using the latest multicanvas renderer file from the repo, as it changed between the test zip and the accepted PR).

The only case I found an issue with was where I'd set the pixelRatio to 10, there was a gap at the end of the waveform:
<img width="512" alt="gap" src="https://cloud.githubusercontent.com/assets/13063119/14235683/7f09d81a-fa48-11e5-9b65-acc451d97ac3.png">

As opposed to before:
<img width="513" alt="orig" src="https://cloud.githubusercontent.com/assets/13063119/14235686/8959dd1a-fa48-11e5-88fb-2c1a5ec1fa96.png">

I'll try to figure it out over the next few days if I get the time.

**Comment by chrisparton1991** [comment]
I've tested the regions plugin example with the `MultiCanvas` renderer and it has no issues. See http://codepen.io/anon/pen/BKmbLw. I forked the example on http://wavesurfer-js.org/plugins/regions.html, updated the library to `1.1.0` and changed the renderer to `MultiCanvas` with `maxCanvasWidth: 50`.

**Comment by chinshr** [comment]
You are right, this example works, since it is not using prerendered peaks. Closest available example is here http://wavesurfer-js.org/example/audio-element/ 

I have it fixed in my app, and I will post the actual lines of code here that made it work.

**Comment by chrisparton1991** [comment]
Ah right, my bad. I forgot about the pre-rendered peaks. I've raised a PR with your fix, can you please check it out? #693 

Of course, if you wanted the fix to be under your name, I'd be happy to close my PR and let you raise one.

**Comment by chrisparton1991** [comment]
Are you happy to have this closed, @chinshr?

**Comment by chinshr** [comment]
This works, fixed in #693.
